game/src/main/scala/io/aigar/game/Player.scala

```diff
@@ -73,6 +74,23 @@ class Player(val id: Int, startPosition: Vector2) extends EntityContainer
     grid.randomPosition
   }
 
+  def merge(cells: List[Cell], player: Player): List[Cell] ={
+    var cellsReturn = List[Cell]()
+
+    for (cell <- cells) {
+      for (secondCell <- cells.filterNot(_.id == cell.id)) {
+        if (cell.id != secondCell.id
+          && cell.overlaps(secondCell)
+          && !cellsReturn.contains(cell)) {
```

Comment:
I agree that there's cleaner functional ways to solve this. But it's acceptable as is